internal/controller/stages/stages.go

```diff
@@ -696,6 +703,36 @@ func (r *reconciler) syncNormalStage(
 
 	status.ObservedGeneration = stage.Generation
 	status.Health = nil // Reset health
+
+	if status.CurrentPromotion != nil {
+		promo, err := r.getPromoFn(ctx, r.kargoClient, types.NamespacedName{
+			Namespace: stage.Namespace,
+			Name:      status.CurrentPromotion.Name,
+		})
+		if err != nil {
+			return status, err
+		}
+
+		freight, err := r.getFreightFn(ctx, r.kargoClient, types.NamespacedName{
+			Namespace: stage.Namespace,
+			Name:      promo.Spec.Freight,
+		})
+		if err != nil {
+			return status, err
+		}
```

Comment:
Can this be copied from `currentPromotion.Freight` instead of being retrieved here?